### PR TITLE
 testing/vault: upgrade to 1.0.3

### PR DESCRIPTION
**Pull request by mtneug**
This upgrades Vault to the latest release.

The current version was quite old and the build process has since been changed significantly. This PR also adds [testing/go-bindata-assetfs](https://github.com/elazarl/go-bindata-assetfs) which is now required for building Vault.

Open notes/questions/issues:

1. Should I add myself as maintainer for go-bindata-assetfs? Since I'm new, just making sure.

2. During the builds, you will see a `loadinternal: cannot find runtime/cgo` warning (only tested on x86_64). I'm not sure if this is specific to Alpine's Go binary or to Vault. However, Vault's Makefile does deactivate CGO by default so this could be normal. The resulting binary seems to work correctly.

### CODE REVIEW

**Comment by Ikke** [comment]
> This upgrades Vault to the latest release.
> 
> The current version was quite old and the build process has since been changed significantly. This PR also adds [testing/go-bindata-assetfs](https://github.com/elazarl/go-bindata-assetfs) which is now required for building Vault.
> 
> Open notes/questions/issues:
> 
> 1. Should I add myself as maintainer for go-bindata-assetfs? Since I'm new, just making sure.

Yes, if you supply the package, you are generally the maintainer as well, unless you find someone else who wants to maintain it.

> 2. During the builds, you will see a `loadinternal: cannot find runtime/cgo` warning (only tested on x86_64). I'm not sure if this is specific to Alpine's Go binary or to Vault. However, Vault's Makefile does deactivate CGO by default so this could be normal. The resulting binary seems to work correctly.

I see that for more packages, so it's not isolated to just Vault. I don't know if it's an issue or not.

File: testing/go-bindata-assetfs/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Contributor: Matthias Neugebauer <mtneug@mailbox.org>
+# Maintainer:
```
Comment (Ikke) [inline testing/go-bindata-assetfs/APKBUILD]: The package should have a maintainer before it can be added.

**Comment by clandmeter** [comment]
please rebase with master so CI will build.

**Comment by mtneug** [comment]
@Ikke thanks for the review. I fixed the issues.

**Comment by Ikke** [comment]
@mtneug APKBUILDs uses tabs instead of spaces. I fixed the issues for now, but please remember for the next time :-)

**Comment by mtneug** [comment]
@Ikke thank you for your help!

**Pull request closed without merging**